docs/source/machine_translation.rst

```diff
@@ -0,0 +1,255 @@
+
+Machine Translation Task Tutorial
+=================================
+
+This tutorial will guide you through designing Machnie Translation Task
+experiments in Neural Monkey. It assumes that you already read the
+Post-Editing tutorial.
+
+The goal of this task is to translate sentences from one language into
+another. For this tutorial we will use data from IT translation task of WMT
+16 on English to Czech direction.
+
+
+Part I. - The Data
+--------------------
+
+We are going to use the data for WMT 16 shared IT Machine Translation task. You
+can get them at the `WMT 16 IT Translation Task
+<http://www.statmt.org/wmt16/it-translation-task.html>`_ and there download
+Batch1 and Batch2 answers and Batch3 as a testing set. Or directly `here
+<http://ufallab.ms.mff.cuni.cz/~popel/batch1and2.zip>`_ and
+`testset <http://ufallab.ms.mff.cuni.cz/~popel/batch3.zip>`_.
+
+Note: Since we are using only small data for example which are not enough
+ for real-life machine translation training.
+
+When we open the archive we find several files for different languages, we
+will use only following files as our training, validation and test set::
+
+    1. ``Batch1a_cs.txt and Batch1a_en.txt`` as our Training set
+    2. ``Batch2a_cs.txt and Batch2a_en.txt`` as a Validation set
+    3. ``Batch3`` as a Test set
+
+Now - before we start, let's make our experiment directory, in which we will
+place all our work. We will call it ``exp-nm-mt``.
+
+First extract all the downloaded files, then make gzip files from individual
+files and put them into following directory structure::
+
+  exp-nm-mt
+  |
+  \== data
+      |
+      |== train
+      |   |
+      |   |== Batch1a_en.txt.gz
+      |   \== Batch1a_cs.txt.gz
+      |
+      |== dev
+      |   |
+      |   |== Batch2a_en.txt.gz
+      |   \== Batch2a_cs.txt.gz
+      |
+      \== test
+          |
+          \== batch3.gz
+
+The gzipping is not a necessary, if you put the dataset there in plaintext it
+ will work the same way. Neural Monkey always try to recognize gzip by the
+extension and opens it.
+
+TODO The dataset is not tokenized and need to be preprocessed.
+
+Byte Pair Encoding
+******************
+
+Neural machine translation (NMT) models typically operate with a fixed
+vocabulary, but translation is an open-vocabulary problem.
+Byte pair encoding (BPE) enables NMT model translation on open-vocabulary by
+encoding rare and unknown words as sequences of subword units.
+This is based on the intuition that various word classes are translatable via
+smaller units than words. More information in the paper
+https://arxiv.org/abs/1508.07909 BPE creates a list of merges that are used
+for splitting out-of-vocabulary words. Example of such splitting::
+
+  basketball => basket@@ ball
+
+Postprocessing can be manually done by::
+
+  sed "s/@@ //g"
```

Comment:
Popsal jsem to tam vice, ze se o to starat nemusi, ale nechal bych tam tuhle zminku